src/main/java/com/aws/greengrass/util/platforms/windows/WindowsExec.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.util.platforms.windows;
+
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+import com.aws.greengrass.util.Utils;
+import com.aws.greengrass.util.platforms.Exec;
+
+import java.io.File;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+@SuppressWarnings("PMD.AvoidCatchingThrowable")
+public class WindowsExec extends Exec {
+    public static final String PATHEXT_KEY = "PATHEXT";
+
+    private static final Logger logger = LogManager.getLogger(WindowsExec.class);
+    private static final List<String> PATHEXT;  // ordered file extensions to try, when no extension is provided
+
+    static {
+        String pathExt = System.getenv(PATHEXT_KEY);
+        PATHEXT = Arrays.asList(pathExt.split(File.pathSeparator));
+    }
+
+    @Nullable
+    @Override
+    public Path which(String fn) {
+        String ext = Utils.extension(fn);
+        // If absolute path, first character expected to be the drive letter
+        if (fn.substring(1).startsWith(":\\")) {
+            if (!ext.isEmpty()) {
+                Path f = Paths.get(fn);
+                return Files.isExecutable(f) ? f : null;
+            }
+            // No extension provided. Try PATHEXT in order
+            for (String extCandidate : PATHEXT) {
+                Path f = Paths.get(fn + extCandidate);
+                if (Files.isExecutable(f)) {
+                    return f;
+                }
+            }
+            return null;
+        }
+        // Search in paths
+        for (Path d : paths) {
```

Comment:
so `paths` still needs to be populated then. Not sure it is getting populated correctly for windows or not

Reply:
Do we have any tests for `which`? Shouldn't be too tough to add. `which("cmd")` `which("cmd.exe")` and so on

Reply:
Exec super class already adds paths from `System.getenv(PATH)`. Added a test case for which